include/deal.II/matrix_free/cuda_matrix_free.templates.h

```diff
@@ -882,29 +900,40 @@ namespace CUDAWrappers
     unsigned int size_co_shape_values =
       n_q_points_1d * n_q_points_1d * sizeof(Number);
 
+    // Check if we already a part of the constant memory allocated to us. If
+    // not, we try to get a block of memory.
+    bool found_id = false;
```

Comment:
can you set my_id to -1 here (or at least assert it)? Can internal_reinit() never happen twice?

Reply:
We set my_id to -1 earlier in the function by calling free.